lib/msf/core/db_manager/http/servlet/workspace_servlet.rb

```diff
@@ -4,10 +4,14 @@ def self.api_path
       '/api/v1/workspaces'
     end
 
+    def self.api_path_with_id
+      "#{WorkspaceServlet.api_path}/?:id?"
+    end
+
     def self.registered(app)
       app.get WorkspaceServlet.api_path, &get_workspace
```

Comment:
We should be using `api_path_with_id` with the get handlers.

Reply:
This is going to take a little bit of rework on the `workspaces` method on the DbManager side to support lookup by `id`, but I'll make the change for consistency's sake.